great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -185,15 +185,16 @@ def configure_validator(self, validator) -> None:
         super().configure_validator(validator)
         validator.expose_dataframe_methods = True
 
-    def load_batch_data(self, batch_id: str, batch_data: Any) -> None:
+    def load_batch_data(
+        self, batch_id: str, batch_data: Union[PandasBatchData, pd.DataFrame]
+    ) -> None:
         if isinstance(batch_data, pd.DataFrame):
             batch_data = PandasBatchData(self, batch_data)
-        elif isinstance(batch_data, PandasBatchData):
-            pass
-        else:
+        elif not isinstance(batch_data, PandasBatchData):
```

Comment:
Not sure if we need this. Depends on the state of type checking in our codebase here.

Reply:
@billdirks This is needed -- enables tests that supply `BatchData` but not formal `Batch` object (similar to an earlier comment).  Thanks.

Reply:
mypy should fail unless we have a `PandasBatchData` or a `pd.DataFrame` object so if mypy is passing I'm not sure we can ender this `raise` codeblock. Are tests that pass mypy hitting this exception?

Reply:
We don't currently run mypy against the tests

Reply:
@billdirks I thought that I fixed this per your feedback (by defining an alias type).  Thanks.